### PR TITLE
Reland "[web] Update text editing test skips"

### PR DESCRIPTION
**Pull request by harryterkelsen**
Reland https://github.com/flutter/engine/pull/37642

Just skip this test on Firefox since it adds random stuff to the style.

### CODE REVIEW

File: lib/web_ui/test/text_editing_test.dart [hunk 1]
```diff
@@ -2579,7 +2578,9 @@ Future<void> testMain() async {
 
       final DomHTMLElement input = editingStrategy!.activeDomElement;
       expect(input.style.getPropertyValue('forced-color-adjust'), 'none');
-    // TODO(hterkelsen): https://github.com/flutter/flutter/issues/115327
+    // TODO(hterkelsen): Firefox does not support forced-color-adjust even
+    // though it supports forced-colors. Safari doesn't support forced-colors
+    // so this isn't a problem there.
```
Comment (yjbanov) [inline lib/web_ui/test/text_editing_test.dart:2583]: Should we still run the test and just alter expectations on different browsers? The prod code will run on all browsers, it just produces different results.

File: lib/web_ui/test/text_editing_test.dart [hunk 2]
```diff
@@ -2565,10 +2565,9 @@ Future<void> testMain() async {
       expect(input.style.backgroundColor, 'transparent');
       expect(input.style.caretColor, 'transparent');
       expect(input.style.outline, 'none');
-      expect(input.style.border, 'none');
+      expect(input.style.border.split(' '), contains('none'));
```
Comment (yjbanov) [inline lib/web_ui/test/text_editing_test.dart:2568]: What's the actual value? I can't tell why the split is necessary. Maybe leave a comment with an explanation?

**Review by yjbanov — APPROVED** [review]

**Comment by skia-gold** [comment]
Gold has detected about 2 new digest(s) on patchset 2.
View them at https://flutter-engine-gold.skia.org/cl/github/37655

**Comment by mdebbar** [comment]
@hterkelsen is this PR still valid?

**Pull request merged**